cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -329,6 +329,23 @@ ARROW_ENGINE_EXPORT Result<DeclarationInfo> MakeAggregateDeclaration(
 
 }  // namespace internal
 
+namespace {
+
+bool IsSortNullsFirst(const substrait::SortField::SortDirection& direction) {
+  return direction % 2 == 1;
```

Comment:
Hmm, rather than arithmetic on enum values, I'd rather see a proper `switch/case` statement to make code more readable and maintainable. This is not so performance-sensitive that it must be optimized to the latest nanosecond.

Reply:
I've switched to proper handling of these cases and I return NullPlacement and SortOrder.  I agree the code is more readable now.